### PR TITLE
Add thin sample model using Jones Matrix

### PR DESCRIPTION
**Pull request by nicolexi**
Add Jones matrix calculation using the model in the paper: https://doi.org/10.1364/BOE.448804

### CODE REVIEW

**Review by diptodip — APPROVED** [review]
@GJBoth LGTM but take a look

File: src/chromatix/elements/polarizer.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+class LinearPolarizer(nn.Module):
+    """
+    Applies a thin polarizer placed after ``Field``.
+    This element returns the ``Field`` directly after the polarizer.
+
+    This element can be placed after any element that returns a ``Field`` or
+    before any element that accepts a ``Field``.
+
+    Attributes:
+        pangle: linear polarizer angle
```
Comment (GJBoth) [inline src/chromatix/elements/polarizer.py:17]: Add the __call__ method before we merge this; or take it out and put it in a separate PR?

Comment (GJBoth) [inline src/chromatix/elements/polarizer.py:17]: Pangle isn't a very clear name - what about `polarisation_angle`?

File: src/chromatix/functional/samples.py [hunk 2]
```diff
@@ -41,3 +41,50 @@ def thin_sample(field: Field, absorption: Array, dn: Array, thickness: Array) ->
...
+        absorption, 6, custom_message="Absorption must be array of shape [1 2 2 H W 1]"
+    )
+    assert_rank(
+        dn, 6, custom_message="Refractive index must be array of shape [1 2 2 H W 1]"
+    )
+
+    # Thickness is the same for four elements in Jones Matrix
+    sample_jones = jnp.exp(1j * 2 * jnp.pi * (dn + 1j * absorption) / field.spectrum)
+    sample_jones = sample_jones[::-1, ::-1]
+    print(sample_jones.shape)
```
Comment (GJBoth) [inline src/chromatix/functional/samples.py:81]: Take out the prints here?

**Review by GJBoth — COMMENTED** [review]
After fixing comments good to merge.

**Review by GJBoth — APPROVED** [review]
LGTM
